src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -201,9 +212,12 @@ public Task SetNormalizedUserNameAsync(IUser user, string normalizedName, Cancel
                 throw new ArgumentNullException(nameof(user));
             }
 
-            ((User)user).NormalizedUserName = normalizedName;
+            if (user is User su)
+            {
+                return Task.FromResult(su.NormalizedUserName);
```

Comment:
Looks like here this is wrong as you don't set the `NormalizedUserName`.

Shoul be

            su.NormalizedUserName = normalizedName;